src/plugins/irc-events/connection.js

```diff
@@ -0,0 +1,75 @@
+var _ = require("lodash");
+var identd = require("../../identd");
+var Msg = require("../../models/msg");
+
+module.exports = function(irc, network) {
+	var client = this;
+
+	client.emit("msg", {
+		chan: network.channels[0].id,
+		msg: new Msg({
+			text: "Network created, connecting to " + network.host + ":" + network.port + "..."
+		})
+	});
+
+	irc.on("raw socket connected", function() {
+		identd.hook(irc.connection.socket, network.username);
```

Comment:
Urgh :-/ (for having something called `connection.socket.socket`, not saying it's not necessary, I don't know, saying however it's confusing)

Then in that case, I'd say let's call `identd.hook(irc.connection.socket.socket, network.username);` and directly use `socket` in the `hook` function. Does that seem reasonable to you?

I know it's close to splitting hairs at this stage, but I'm just trying to close the loop :-)